instance_clusters.go

```diff
@@ -0,0 +1,206 @@
+	PlatformKubernetes *PlatformKubernetes `json:"platform_kubernetes"`
+	ManagementProject  *ManagementProject  `json:"management_project"`
+}
+
+func (v InstanceCluster) String() string {
+	return Stringify(v)
+}
+
+// PlatformKubernetes represents a GitLab Instance Cluster PlatformKubernetes.
+type InstanceClusterPlatformKubernetes struct {
+	APIURL            string `json:"api_url"`
+	Token             string `json:"token"`
+	CaCert            string `json:"ca_cert"`
+	Namespace         string `json:"namespace"`
+	AuthorizationType string `json:"authorization_type"`
+}
+
+// ListClusters gets a list of all instance clusters.
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ee/api/instance_clusters.html#list-instance-clusters
+func (s *InstanceClustersService) ListClusters(options ...RequestOptionFunc) ([]*InstanceCluster, *Response, error) {
+	u := fmt.Sprintf("admin/clusters")
+
+	req, err := s.client.NewRequest("GET", u, nil, options)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var ics []*InstanceCluster
+	resp, err := s.client.Do(req, &ics)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return ics, resp, err
+}
+
+// GetCluster gets an instance cluster.
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ee/api/instance_clusters.html#get-a-single-instance-cluster
+func (s *InstanceClustersService) GetCluster(clusterID int, options ...RequestOptionFunc) (*InstanceCluster, *Response, error) {
+	u := fmt.Sprintf("admin/clusters/%d", clusterID)
+
+	req, err := s.client.NewRequest("GET", u, nil, options)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	ic := new(InstanceCluster)
+	resp, err := s.client.Do(req, &ic)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return ic, resp, err
+}
+
+// AddInstanceClusterOptions represents the available AddCluster() options.
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ee/api/instance_clusters.html#add-existing-cluster-to-instance
+type AddInstanceClusterOptions struct {
+	Name                *string                               `url:"name,omitempty" json:"name,omitempty"`
+	Domain              *string                               `url:"domain,omitempty" json:"domain,omitempty"`
+	Enabled             *bool                                 `url:"enabled,omitempty" json:"enabled,omitempty"`
+	Managed             *bool                                 `url:"managed,omitempty" json:"managed,omitempty"`
+	EnvironmentScope    *string                               `url:"environment_scope,omitempty" json:"environment_scope,omitempty"`
+	PlatformKubernetes  *AddInstancePlatformKubernetesOptions `url:"platform_kubernetes_attributes,omitempty" json:"platform_kubernetes_attributes,omitempty"`
+	ManagementProjectID *string                               `url:"management_project_id,omitempty" json:"management_project_id,omitempty"`
+}
+
+// AddInstancePlatformKubernetesOptions represents the available PlatformKubernetes options for adding.
+type AddInstancePlatformKubernetesOptions struct {
```

Comment:
```suggestion
type PlatformKubernetesOptions struct {
```

Reply:
I think `AddPlatformKubernetesOptions` would be more explicit here, since these are the options for adding a cluster. But then `AddPlatformKubernetesOptions` is already declared in https://github.com/xanzy/go-gitlab/blob/d3946d4f8c62bfae945062be289f13ea5e6ecfe4/project_clusters.go#L140 so the compiler throws a redeclaration error

Reply:
Ah, I just saw your note on reusing `PlatformKubernetesOptions` for both Add and Edit cluster. In that case, we don't name this `AddPlatformKubernetesOptions` and there is not a redeclaration. Making the change!